### PR TITLE
[FIX] fix error PTH106 

### PR DESCRIPTION
**Pull request by hndgzkn**
- Closes  none but relates to #4327 

Changes proposed in this pull request:

- fixes error: https://docs.astral.sh/ruff/rules/os-rmdir/

### CODE REVIEW

File: nilearn/datasets/tests/test_utils.py [hunk 1]
```diff
@@ -244,43 +244,45 @@ def test_tree():
...
+    dir2 = tmp_path / "dir2"
+    dir212 = dir2 / "dir12"
+
+    dir1.mkdir()
+    dir111.mkdir()
+    dir112.mkdir()
+    dir2.mkdir()
+    dir212.mkdir()
+
+    open(dir1 / "file11", "w").close()
```
Comment (Remi-Gau) [inline nilearn/datasets/tests/test_utils.py:261]:
```suggestion
    (dir1 / "file11").touch()
```

maybe double check but I think that Path has a `touch` method that should allow to simplify those open/close even further.

File: nilearn/datasets/_utils.py [hunk 2]
```diff
@@ -702,24 +702,24 @@ def movetree(src, dst):
 
     Any existing file is overwritten.
     """
-    names = os.listdir(src)
```
Comment (Remi-Gau) [inline nilearn/datasets/_utils.py:705]: I am wondering if we even need this movetree function.

Not sure how it differs
https://docs.python.org/3/library/shutil.html#shutil.move

@hndgzkn can you double check and maybe update the doc string to explain why it is different?

Comment (Remi-Gau) [reply nilearn/datasets/_utils.py:705]: thanks a bunch: I and future maintainers thanks you for the clarification.

Comment (hndgzkn) [reply nilearn/datasets/_utils.py:705]: I hope it is clear enough.

**Review by Remi-Gau — APPROVED** [review]
looks good to me: will @michellewang cross check

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Review by michellewang — APPROVED** [review]
Looks good to me as well. There will probably be a merge conflict with #4606 in `test_utils.py` but it should be easy to resolve :)

**Pull request merged**